### PR TITLE
Added arm, s390x and ppc64le support for nodejs

### PR DESCRIPTION
**Pull request by omBratteng**
This builds on the PR #1105

### CODE REVIEW

**Comment by omBratteng** [comment]
Should we add support for these architectures? I personally don't use these architectures. But looking at [s390x/node](https://hub.docker.com/r/s390x/node) it has 1M+ pulls the last month, and [ppc64le/node](https://hub.docker.com/r/ppc64le/node) has 10M+ pulls.
So someone is using it.

**Comment by omBratteng** [comment]
@loosebazooka Should we add support for these architectures? I personally don't use these architectures. But looking at [s390x/node](https://hub.docker.com/r/s390x/node) it has 1M+ pulls the last month, and [ppc64le/node](https://hub.docker.com/r/ppc64le/node) has 10M+ pulls.
So someone is using it.

**Comment by loosebazooka** [comment]
I think it's fine to include then. I'm on parental leave right now so I'm pretty slow, but I'll find some time to take a look when the pr is out of draft.

**Comment by omBratteng** [comment]
Shouldn't [this line](https://github.com/GoogleContainerTools/distroless/pull/1317/files#diff-3053677e4333a96b20ed76c662bca07fb2e1ce059479170139209356787da09cR220) be `NODEJS |= {`?
The [line 211 above](https://github.com/GoogleContainerTools/distroless/pull/1317/files#diff-3053677e4333a96b20ed76c662bca07fb2e1ce059479170139209356787da09cR211) already is `NODEJS = {`

File: BUILD [hunk 1]
```diff
@@ -210,23 +210,23 @@ NODEJS_VARIATIONS = [
...
-    for arch in BASE_ARCHITECTURES
-    for distro in ARCHITECTURES
+    for arch in ARCHITECTURES
+    for distro in LANGUAGE_DISTROS
     for version in NODEJS_VERSIONS
     for (tag_base, label, user) in NODEJS_VARIATIONS
 }
 
 # oci_image_index
 NODEJS = {
```
Comment (omBratteng) [inline BUILD]: Shouldn't this be `NODEJS |= {`?
The line 211 above already is `NODEJS = {`

Comment (loosebazooka) [reply BUILD]: yeah probably.

**Comment by omBratteng** [comment]
For some reason, when I add a 2nd new architecture, it fails to fetch/ build some java dependencies. Scratching my head to as why that happens

**Comment by loosebazooka** [comment]
Weird. Maybe we're hitting some size limit on something.

**Comment by omBratteng** [comment]
I think I am running into same problem locally when I try to build all, but unsure if I might have issues because I am on apple arm laptop. I'll try on another machine to test it out locally

**Comment by omBratteng** [comment]
Just ran this in a VM with 240 GB available disk, but used LVM to start at 20 GB.
Had to bump the disk up to 40 GB before the bazel build command worked. And another two for the tests to work.
So it looks like this is caused by too little available diskspace in the workflow.

I wonder if it could be solved by separating the builds by arch with a workflow matrix.
The `CI Image tests` job only runs `amd64` via a query.
https://github.com/GoogleContainerTools/distroless/blob/3886376f235f5628387492a903093ebeb4fcdaa0/.github/workflows/ci.yaml#L71
So I was thinking if it would be possible to combine that query with this one
https://github.com/GoogleContainerTools/distroless/blob/3886376f235f5628387492a903093ebeb4fcdaa0/.github/workflows/ci.yaml#L42

**Comment by omBratteng** [comment]
I think I got it. [Running with this query `bazel query 'kind(oci_image, attr("name", "${{ matrix.arch }}", deps(//:sign_and_push)))'`](https://github.com/omBratteng/distroless/actions/runs/5204908442) in my repo right now. I'll give it some time and see if it works.

Example output from that query
```
❯ bazel query 'kind(oci_image, attr("name", "arm64", deps(//:sign_and_push)))'                                                                                                          

//base:base_nonroot_arm64_debian11
//base:base_nossl_debug_nonroot_arm64_debian11
//base:base_nossl_debug_root_arm64_debian11
//base:base_nossl_nonroot_arm64_debian11
//base:base_nossl_root_arm64_debian11
//base:base_root_arm64_debian11
//base:debug_nonroot_arm64_debian11
//base:debug_root_arm64_debian11
//base:static_debug_nonroot_arm64_debian11
//base:static_debug_root_arm64_debian11
//base:static_nonroot_arm64_debian11
//base:static_root_arm64_debian11
//cc:cc_nonroot_arm64_debian11
//cc:cc_root_arm64_debian11
//cc:debug_nonroot_arm64_debian11
//cc:debug_root_arm64_debian11
//experimental/python3:debug_nonroot_arm64_debian11
//experimental/python3:debug_root_arm64_debian11
//experimental/python3:python3_nonroot_arm64_debian11
//experimental/python3:python3_root_arm64_debian11
//java:java11_debug_nonroot_arm64_debian11
//java:java11_debug_root_arm64_debian11
//java:java11_nonroot_arm64_debian11
//java:java11_root_arm64_debian11
//java:java17_debug_nonroot_arm64_debian11
//java:java17_debug_root_arm64_debian11
//java:java17_nonroot_arm64_debian11
//java:java17_root_arm64_debian11
//java:java_base_debug_nonroot_arm64_debian11
//java:java_base_debug_root_arm64_debian11
//java:java_base_nonroot_arm64_debian11
//java:java_base_root_arm64_debian11
//nodejs:nodejs16_debug_nonroot_arm64_debian11
//nodejs:nodejs16_debug_root_arm64_debian11
//nodejs:nodejs16_nonroot_arm64_debian11
//nodejs:nodejs16_root_arm64_debian11
//nodejs:nodejs18_debug_nonroot_arm64_debian11
//nodejs:nodejs18_debug_root_arm64_debian11
//nodejs:nodejs18_nonroot_arm64_debian11
//nodejs:nodejs18_root_arm64_debian11
//nodejs:nodejs20_debug_nonroot_arm64_debian11
//nodejs:nodejs20_debug_root_arm64_debian11
//nodejs:nodejs20_nonroot_arm64_debian11
//nodejs:nodejs20_root_arm64_debian11
```

**Comment by omBratteng** [comment]
I see the output does not contain the `sign_and_push_*_sbom` targets. Not sure if they are required for CI?

**Comment by loosebazooka** [comment]
CI has been updated to add a bunch of diskspace, any chance of a rebase on this? I assume we can just hold off on doing this for debian11, and make it available on debian12.

**Comment by omBratteng** [comment]
Just rebased, will change from debian 11 to 12

**Comment by omBratteng** [comment]
Guess I could call the ready for review. Even though these are not targeted for debian 11, they still get built. Should I add it in the readme?

**Comment by loosebazooka** [comment]
Ah didn't see you said this is easy. Will take a look Monday

**Review by loosebazooka — APPROVED** [review]

**Comment by loosebazooka** [comment]
looks good, lets just rebase and merge?

**Comment by omBratteng** [comment]
Rebased

**Comment by loosebazooka** [comment]
thanks @omBratteng, you contributions to keeping nodejs on distroless up to date are much appreciated!

**Pull request merged**